core/src/utilities/helpers/routing-helpers.js

```diff
@@ -162,16 +162,12 @@ class RoutingHelpersClass {
       }
     });
 
-    if (hashRoutingActive) {
-      return EventListenerHelpers.addEventListener('hashchange', event => {
-        const withoutSync = EventListenerHelpers.hashChangeWithoutSync;
-        EventListenerHelpers.hashChangeWithoutSync = false;
-        callback(Routing.getHashPath(event.newURL), withoutSync);
-      });
-    }
-
     EventListenerHelpers.addEventListener('popstate', e => {
-      callback(Routing.getModifiedPathname(), e.detail && e.detail.withoutSync);
+      if (hashRoutingActive) {
+        callback(Routing.getHashPath(location.href), e.detail && e.detail.withoutSync);
+      } else {
+        callback(Routing.getModifiedPathname(), e.detail && e.detail.withoutSync);
+      }
```

Comment:
```suggestion



      const method = hashRoutingActive ? Routing.getHashPath(location.href): Routing.getModifiedPathname();
       callback(method, e.detail && e.detail.withoutSync);
```
This way is cleaner. `method` just an example, please feel free to change the naming.

Reply:
@JohannesDoberer, please apply this improvement. The rest looks good.